core/src/main/java/com/linecorp/armeria/server/Http2RequestDecoder.java

```diff
@@ -394,7 +406,12 @@ public void onRstStreamRead(ChannelHandlerContext ctx, int streamId, long errorC
 
         final ClosedStreamException cause =
                 new ClosedStreamException("received a RST_STREAM frame: " + Http2Error.valueOf(errorCode));
-        req.abortResponse(cause, /* cancel */ true);
+        if (req.needsAggregation() && !req.isInitialized()) {
```

Comment:
```suggestion
        if (req.needsAggregation()) {
            assert !req.isInitialized();
```

Reply:
This isn't true because the server can receive an RST_STREAM after the aggregating request is initialized.

Reply:
> request is initialized.

Q: Initialization seems more important to call `req.toAbortedStreaming()`. What do you think of checking `req.isInitialized()` where a call to `req.toAbortedStreaming()` is necessary for here and L335?
```java
if (!req.isInitialized()) {
   ctx.fireChannelRead(req.toAbortedStreaming(inboundTrafficController, cause, false));
}
```

Reply:
That's a good suggestion. Let me change it back. 😉

Reply:
The original reason for my suggestion was because I wanted to do some testing on rescheduling the layer between channel and service event loops.

Let me just handle this separately then if needed